### PR TITLE
Added `is_secure` column to `query_log`

### LINKED ISSUE #34288: Add HTTP-SECURE and NATIVE-SECURE Interface entries to system.query_log
**Use case**
We're currently in the process of migrating all clients over from connecting to Native and HTTP endpoints (non TLS) over to secure  endpoints using TLS.

As part of this, given we have thousands of different clients connecting to multiple different clusters, we would like to  have metrics on the clients that are still making connections to the unencrypted endpoints.

Looking at `system.query_log` (https://clickhouse.com/docs/en/operations/system-tables/query_log/) , I don't see anything in there that allows us to distinguish between connections made on secure or insecure ports. (e.g 9000 instead of 9440)

I also don't see anything in any other tables that have this information (e.g `system.processes`), but please do let me know if I have missed anything.

**Describe the solution you'd like**

After looking at other tables and the `system.query_log`, I can see we have an `Interface` column. that contains HTTP and NATIVE as possible values. It would be great if we could add another 2 options here for HTTP-SECURE and NATIVE-SECURE, and set these options accordingly if the connection established with the client uses TLS, or one of the secure ports. e.g 9440.

That way, we could do things like:

```
SELECT
    initial_user,
    client_name,
    query
FROM system.query_log
WHERE (event_date = today()) AND (interface = 1)
GROUP BY initial_user, client_name, query
```

and know what connections we'll break if we turn the insecure endpoints off.

**Describe alternatives you've considered**

I have looked at other tables, and also looked at trying to obtain this information outside of ClickHouse replicas, but it feels messy and would be difficult to get information that is readily available in the query_log. e.g initial_user.

ClickHouse version:  v21.8

Thank you in advance, and I am happy to do the work for this issue if you are happy with the proposal. Please let me know if I can provide any more information!

### PR DESCRIPTION
**Pull request by antonio2368**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added `is_secure` column to `system.query_log` which denotes if the client is using a secure connection over TCP or HTTP.

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: tests/queries/0_stateless/02246_is_secure_query_log.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
```
Comment (kssenii) [inline tests/queries/0_stateless/02246_is_secure_query_log.sh:1]: Looks like need a tag: `Tags: no-fasttest`

**Review by kssenii — APPROVED** [review]

File: tests/queries/0_stateless/02246_is_secure_query_log.sh (hunk 1 above)
Comment (antonio2368) [reply tests/queries/0_stateless/02246_is_secure_query_log.sh:1]: Yep

**Comment by antonio2368** [comment]
@Mergifyio update

**Comment by CAJan93** [comment]
Could we add this to the official docs? https://clickhouse.com/docs/operations/system-tables/query_log

**Pull request merged**